com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/DebugAdapter.java

```diff
@@ -113,6 +116,19 @@ private void initialize() {
     }
 
     private void registerHandler(IDebugRequestHandler handler) {
+        registerHandler(handler, true, false);
+    }
+
+    private void registerHandler(IDebugRequestHandler handler, boolean forDebug, boolean forRun) {
+        if (forDebug) {
+            registerHandler(requestHandlersForDebug, handler);
+        }
+        if (forRun) {
+            registerHandler(requestHandlersForRun, handler);
```

Comment:
I don't think parameter forDebug forRun will help to understand the handlers, I suguest to have two registerHandlerForDebug and registerHandlerForNonDebug private helper methods.

Reply:
Agree. Naming is always a tough thing. Since the flag is named `noDebug`, here should I use `registerHandlerForNonDebug` or `registerHandlerForNoDebug`. @andxu @testforstephen

Reply:
Keep the consistent. I like `registerHandlerforNoDebug`

Reply:
About naming, here is a good suggestion.

`
Pro Tip: don't try and name a function by looking at its implementation as you'll only come up with a name that reflects how it works. Instead look at the call sites and see what they want to know or do.
`